src/main/java/graphql/analysis/QueryVisitorStub.java

```diff
@@ -0,0 +1,23 @@
+package graphql.analysis;
+
+import graphql.PublicApi;
+
+@PublicApi
+public class QueryVisitorStub implements QueryVisitor {
```

Comment:
Another way to do this in Java 8 is just use default methods on the interface.

I am still 50/50 on whether this is a good idea.

I do like your naming here.  Better than NoopFoo